actions/class.ItemContent.php

```diff
@@ -202,7 +203,13 @@ private function getMediaAsset(bool $validateWriteAccess = false): ?MediaAsset
     {
         $params = $this->getRequiredQueryParams('uri', 'lang', 'path');
         $asset = $this->resolveAsset($params['uri'], $params['path'], $params['lang']);
-        $resourceUri = tao_helpers_Uri::decode($asset->getMediaIdentifier());
+
+        $mediaSource = $asset->getMediaSource();
+        if ($mediaSource instanceof MediaSource) {
```

Comment:
You have some options here, maybe: 

- Check based on the path you received.
- Check if they are instance of `HttpSource` or `LocalItemSource` instead.